### PR TITLE
Add dataSourceViewId in agent_data_source_configurations

### PR DESCRIPTION
**Pull request by flvndvd**
## Description

This PR introduces the `dataSourceViewId` column to the `agent_data_source_configurations` table. Currently, `dataSourceViewId` is inferred from the selected data source, defaulting to the view within the workspace vault, and is not passed in the API call yet. This is a temporary solution to start transitionning our logic to data source views. Additionally, we need to update the existing index in `data_source_views` to include a composite index on `workspaceId`, `dataSourceId`, and `vaultId`.

Fow now, we also restrict to one data source view per data source per vault. 

## Deploy Plan
- [x] Run the SQL migration

### CODE REVIEW

File: front/lib/api/assistant/configuration.ts [hunk 1]
```diff
@@ -1504,6 +1509,24 @@ async function _createAgentDataSourcesConfigData(
...
+  const dataSourceViews = (
+    await Promise.all(
+      uniqueDataSources.map((ds) => {
+        return DataSourceViewResource.listForDataSourceInVault(
+          auth,
+          renderDataSourceType(ds),
+          globalVault,
+          { transaction: t }
+        );
+      })
```
Comment (tdraier) [inline front/lib/api/assistant/configuration.ts:1526]: Can't we do a single query by passing a list of datasources (datasourceIds) ? `listForDataSourcesInVault`

Comment (flvndvd) [reply front/lib/api/assistant/configuration.ts]: Done.

File: front/lib/api/assistant/configuration.ts [hunk 2]
```diff
@@ -1504,6 +1509,19 @@ async function _createAgentDataSourcesConfigData(
...
   const dataSources = results.flat();
 
+  // Since the UI does not currently provide the data source view,
+  // we retrieve the view associated with the data from the global vault
+  // and assign it to the agent data source configuration.
+  const uniqueDataSources = _.uniqBy(dataSources, (ds) => ds.id);
+  const globalVault = await VaultResource.fetchWorkspaceGlobalVault(auth);
+  const dataSourceViews =
+    await DataSourceViewResource.listForDataSourcesInVault(
+      auth,
```
Comment (flvndvd) [inline front/lib/api/assistant/configuration.ts:1519]: Open a thread [here](https://dust4ai.slack.com/archives/C050SM8NSPK/p1722357112317509) to understand why we can have several workspace ids.

Comment (spolu) [reply front/lib/api/assistant/configuration.ts:1519]: Responded there 👍

**Comment by spolu** [comment]
This PR assumes a view for all data source which is not what was planned as discussed https://dust4ai.slack.com/archives/C050SM8NSPK/p1722368349079209?thread_ts=1722332810.418999&cid=C050SM8NSPK

Let's converge on this discussion before moving forward here.

**Comment by flvndvd** [comment]
> This PR assumes a view for all data source which is not what was planned as discussed https://dust4ai.slack.com/archives/C050SM8NSPK/p1722368349079209?thread_ts=1722332810.418999&cid=C050SM8NSPK
> 
> Let's converge on this discussion before moving forward here.

Updated based on the discussion.

**Review by tdraier — APPROVED** [review]

**Pull request merged**